src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -95,30 +89,138 @@ public static Email parseEmail(String email) throws ParseException {
+        case "TRACK MEN" -> Sport.TRACK_M;
+        case "TRACK WOMEN" -> Sport.TRACK_W;
+        case "ULTIMATE FRISBEE" -> Sport.ULTIMATE_FRISBEE;
+        case "VALORANT" -> Sport.VALORANT;
+        case "VOLLEYBALL MEN" -> Sport.VOLLEYBALL_M;
+        case "VOLLEYBALL WOMEN" -> Sport.VOLLEYBALL_W;
+        default -> throw new ParseException("Invalid sport: " + sport);
+        };
+    }
+
+    private static Branch parseBranch(String branch) throws ParseException {
+        return switch (branch.toUpperCase()) {
+        case "MARKETING" -> Branch.MARKETING;
+        case "PUBLICITY" -> Branch.PUBLICITY;
+        case "SPORTS" -> Branch.SPORTS;
+        default -> throw new ParseException("Unexpected branch: " + branch);
+        };
+    }
+
+    private static Position parsePosition(String position) throws ParseException {
+        return switch (position.toUpperCase()) {
+        case "PROJECT DIRECTOR" -> Position.PROJECT_DIRECTOR;
+        case "VICE PROJECT DIRECTOR" -> Position.VICE_PROJECT_DIRECTOR;
+        case "SPORTS DIRECTOR" -> Position.SPORTS_DIRECTOR;
+        case "VICE SPORTS DIRECTOR" -> Position.VICE_SPORTS_DIRECTOR;
+        case "MEMBER" -> Position.MEMBER;
+        default -> throw new ParseException("Unexpected position: " + position);
+        };
+    }
+
+    private static VolunteerRole parseVolunteer(String volunteer) throws ParseException {
+        return switch(volunteer.toUpperCase()) {
+        case "PHOTOGRAPHER" -> VolunteerRole.PHOTOGRAPHER;
+        case "EMCEE" -> VolunteerRole.EMCEE;
+        case "USHER" -> VolunteerRole.USHER;
+        case "LOGISTICS" -> VolunteerRole.LOGISTICS;
+        case "FIRST AID" -> VolunteerRole.FIRST_AID;
+        case "BOOTH MANNER" -> VolunteerRole.BOOTH_MANNER;
+        default -> throw new ParseException("Unexpected volunteer role: " + volunteer);
+        };
+    }
+
     /**
-     * Parses a {@code String tag} into a {@code Tag}.
+     * Parses a {@code String role} into a {@code Role}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code tag} is invalid.
+     * @throws ParseException if the given {@code role} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
-        requireNonNull(tag);
-        String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+    public static Role parseRole(String role) throws ParseException {
+        requireNonNull(role);
+        String trimmedRole = role.trim();
+        if (!Role.isValidRoleName(trimmedRole)) {
+            throw new ParseException(Role.MESSAGE_CONSTRAINTS);
+        }
+        String[] tagSplit = trimmedRole.split(" - ");
+        String roleType = tagSplit[0];
+        return switch (roleType.toLowerCase()) {
+        case "athlete" -> {
+            Faculty faculty = parseFaculty(tagSplit[1]);
```

Comment:
I'm assuming error handling needs to be done here in case tagSplit[1] is, for example, not a faculty?!

Reply:
Using ParseExceptions shown to the CLI for now. When an add command with improper format is entered, ParserUtil should be calling ParseExceptions to abort the parsing of the command.